azavea/echo-locator
# Handle voucher/no voucher cases in EditProfile

## Overview

This PR adds conditional logic to display different questions in the `EditProfile` component based on whether or not the user has a voucher.


### Checklist


### Demo

![Screen Shot 2022-05-16 at 1 47 08 PM](https://user-images.githubusercontent.com/77936689/168655159-54c3efd8-540f-47b3-8879-fd9be5487dca.png)

If a user _does_ have a voucher, they will see the "voucher number" and "number of bedrooms on your voucher" fields.

![Screen Shot 2022-05-16 at 1 47 00 PM](https://user-images.githubusercontent.com/77936689/168655169-6a001824-e6dc-4f45-8861-041d1acaf19a.png)

 If they _do not_ have a voucher, they will see the "budget for apartment" and "desired number of bedrooms" fields.


### Notes

I initially decided to use radio buttons as opposed to a checkbox for the "Do you have a voucher" question because that was what the demo site uses. I also spoke with @gatesgodin, who verified that the radio buttons were appropriate here.

Currently, if a user switches from being a voucher user to not (or vice versa), the database will still keep the previously entered values, e.g. if a user who initially had a voucher filled out their voucher number and number of bedrooms then switched to not having a voucher and filled out their desired number of bedrooms and rent budget, the database would keep all 4 values and would not clear the voucher-related values following the switch.


## Testing Instructions

 * Run `./scripts/server`
 * Navigate to **localhost:9966** and login
 * Click "Edit profile" and verify that changing your answer to "Do you have a voucher?" changes the fields as illustrated above
 * Make a change to one of the altered fields and verify that it successfully saves to the database


Resolves #461
